lib/theme_check/template.rb

```diff
@@ -31,15 +40,17 @@ def snippet?
       name.start_with?('snippets')
     end
 
-    def source
-      @source ||= @path.read
-    end
-
     def lines
       # Retain trailing newline character
       @lines ||= source.split("\n", -1)
     end
 
+    # Not entirely obvious but lines is mutable, corrections are to be
```

Comment:
Instead we should just reset `@lines` in `write`, this way `lines` will always be the correct method to call:

```ruby
def write
  # ...
  @lines = nil
end
```

Reply:
Not sure I understand this one. The intention behind this method was to do a check on source != updated_source before writing. And to use it in tests rather than the path.read we had in test_helper.